src/app/pages/vm/vm-wizard/vm-wizard.component.ts

```diff
@@ -874,9 +880,19 @@ async customSubmit(value) {
         {"dtype": "DISK", "attributes": {"path": hdd, "type": value.hdd_type, 'physical_sectorsize': null, 'logical_sectorsize': null}},
       ]
     }
-    
-    if(value.enable_vnc &&value.bootloader !== "UEFI_CSM"){
-      vm_payload["devices"].push({
+
+    if (value.enable_vnc) {
+      if (this.productType === 'SCALE') {
+        vm_payload["devices"].push({
+          "dtype": "VNC", "attributes": {
+            "vnc_port": String(this.getRndInteger(5900,65535)),
```

Comment:
I just noticed, is this a custom function retrieve a random integer between this range ?
If yes, we should not be doing this and instead we should make a call to middleware which will return the next available vnc port `vm.vnc_port_wizard`.